ethcore/engines/authority-round/src/lib.rs

```diff
@@ -1430,6 +1452,46 @@ impl Engine for AuthorityRound {
 		block_reward::apply_block_rewards(&rewards, block, &self.machine)
 	}
 
+	/// Make calls to the randomness contract.
+	fn on_prepare_block(&self, block: &ExecutedBlock) -> Result<Vec<SignedTransaction>, Error> {
```

Comment:
> Is it still guaranteed at this point that the client's latest nonce is the same, i.e. that the latest block as seen by the client is the parent of this `block`?

I honestly do not know. @tomusdrw maybe?

Reply:
Yes, the `nonce` should be the same since we are only calling this function before any transaction is applied. I think we should actually consider renaming this function to `fn generate_engine_transactions(&self, block: &ExecutedBlock) -> Result<Vec<_>>` and document it properly. Having `on_prepare_block` return a list of transactions doesn't seem intuitive for me.

Reply:
I renamed it.
There is a doc comment on the method in the trait definition.